src/components/MetadataManager.js

```diff
@@ -0,0 +1,224 @@
+// @flow
+import React from 'react';
+import PropTypes from 'prop-types';
+import { withRouter } from 'react-router-dom';
+
+import QueryResponse from '../api/QueryResponse';
+import SimpleQueryRequest from '../api/SimpleQueryRequest';
+import SavedSearch from '../api/SavedSearch';
+import AuthUtils from '../util/AuthUtils';
+import Configurable from '../components/Configurable';
+
+type MetadataManagerProps = {};
+
+type MetadataManagerDefaultProps = {};
+
+type MetadataManagerState = {
+  response?: QueryResponse,
+  error?: string,
+  savedSearchList: Array<SavedSearch>,
+};
+/**
+ * Component that manages all the metadata management, manages functions that require api calls
+ * but are not relevant to be put in search
+ */
+class MetadataManager extends React.Component<MetadataManagerDefaultProps, MetadataManagerProps, MetadataManagerState> {
+  static defaultProps: MetadataManagerDefaultProps = {};
+  static childContextTypes = {
+    metadataManager: PropTypes.any,
+  };
+
+  static displayName = 'MetadataManager';
+
+  constructor(props: MetadataManagerProps) {
```

Comment:
Unnecessary constructor. Prefer arrow functions over function bindings. State is not based on props. Set in line 47. Note that if you do not set an initial state value for a state const, that const will be `undefined` by default. Prefer `null` over `undefined`.